crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -2344,6 +2344,352 @@ pub fn local_equivalence(weyl: PyReadonlyArray1<f64>) -> PyResult<[f64; 3]> {
+            }
+        }
+
+        Ok(TwoQubitGateSequence {
+            gates,
+            global_phase,
+        })
+    }
+
+    /// Appends U_d(a, b, c) to the circuit.
+    fn weyl_gate(
+        &self,
+        circ: &mut TwoQubitGateSequence,
+        target_decomposed: TwoQubitWeylDecomposition,
+        atol: f64,
+    ) -> PyResult<()> {
+        let circ_a = self.to_rxx_gate(-2.0 * target_decomposed.a)?;
+        circ.gates.extend(circ_a.gates);
+        let mut global_phase = circ_a.global_phase;
+
+        if (target_decomposed.b).abs() > atol {
+            let circ_b = self.to_rxx_gate(-2.0 * target_decomposed.b)?;
+            global_phase += circ_b.global_phase;
+            circ.gates
+                .push((Some(StandardGate::SdgGate), smallvec![], smallvec![0]));
+            circ.gates
+                .push((Some(StandardGate::SdgGate), smallvec![], smallvec![1]));
+            circ.gates.extend(circ_b.gates);
+            circ.gates
+                .push((Some(StandardGate::SGate), smallvec![], smallvec![0]));
+            circ.gates
+                .push((Some(StandardGate::SGate), smallvec![], smallvec![1]));
+        }
+
+        if (target_decomposed.c).abs() > atol {
+            let mut gamma = -2.0 * target_decomposed.c;
+            let mut invert = false;
+            if gamma > 0.0 {
+                gamma *= -1.0;
+                invert = true;
+            }
+            let circ_c = self.to_rxx_gate(gamma)?;
+            if !invert {
+                global_phase += circ_c.global_phase;
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![0]));
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![1]));
+                circ.gates.extend(circ_c.gates);
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![0]));
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![1]));
+            } else {
+                // invert the circuit above
+                global_phase -= circ_c.global_phase;
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![0]));
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![1]));
+                for gate in circ_c.gates.into_iter().rev() {
+                    let (inv_gate_name, inv_gate_params, inv_gate_qubits) = invert_2q_gate(gate);
+                    circ.gates
+                        .push((Some(inv_gate_name), inv_gate_params, inv_gate_qubits));
+                }
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![0]));
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![1]));
+            }
```

Comment:
I would be slightly inclined to simplify the logic here to remove the `invert` variable and only have one if statement on `gamma > 0.0`.

Reply:
Done in [9ac68b1](https://github.com/Qiskit/qiskit/pull/13139/commits/9ac68b1d1ead2d8ae18be96732ecf2b6b2af638d)